src/plugins/kibana/public/settings/sections/advanced/index.js

```diff
@@ -17,39 +16,38 @@ uiModules.get('apps/settings')
   return {
     restrict: 'E',
     link: function ($scope) {
-      const configDefaults = Private(ConfigDefaultsProvider);
       const keyCodes = {
         ESC: 27
       };
 
-      function isTypeComplex(conf) {
-        return !(conf.json || conf.array || conf.bool || conf.normal);
-      }
-
-      function notDefaultConfig(configName) {
-        return !(configName in configDefaults);
-      }
-
-      function readConfigVals() {
-        const configVals = config._vals();
-
-        const customConfig = Object.keys(configVals)
-        .filter(notDefaultConfig)
-        .map(name => toEditableConfig(false, name, configVals[name]));
-
-        $scope.configs = _(configDefaults)
-        .map((def, name) => toEditableConfig(def, name, configVals[name]))
-        .reject('readonly')
-        .concat(customConfig)
-        .value();
-      }
-
       // react to changes of the config values
-      const unhook = $rootScope.$on('change:config', readConfigVals);
+      const unhook = $rootScope.$on('change:config', changed);
```

Comment:
This also raises an interesting point: changed `config.on` to unhook listeners on `scope#$destroy`